### PR TITLE
DTSSTCI-822: DAC - Redundant Entry (Submit Case)

### PR DESCRIPTION
**Pull request by pr00279**
### JIRA link

https://tools.hmcts.net/jira/browse/DTSSTCI-822

### Change description
This is the Submit Case counterpart to DTSSTCI-725 (hence the branch name). Displays the document relevance alongside each file name. Also does not erase input to "Additional information" textbox on file upload.

Changes to 'Add information to a case' page (also added some <br> tags to space out the additional info and the document relevance and upload from each other):
image

Changes to 'Check your answers' page:
Pasted Graphic 1

UPDATED so that the headers are inline:

image

ADDITIONAL UPDATE such that 'Document relevance' appears inline with the document relevance value:

image

Case Documents tab in Case API:
Pasted Graphic 3

Messages tab in Case API:
Pasted Graphic 4

### CODE REVIEW

File: charts/sptribs-frontend/Chart.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ appVersion: '1.0'
 description: A Helm chart for sptribs-frontend App
 name: sptribs-frontend
 home: https://github.com/hmcts/sptribs-frontend
-version: 0.0.35
+version: 0.0.36
```
Comment (gmmagruder) [inline charts/sptribs-frontend/Chart.yaml:6]: Think this was bumped when you changed changed the case api url. Make sure to revert this back before merge.

File: src/main/resources/check-your-answers/translation.json [hunk 2]
```diff
@@ -27,7 +27,7 @@
         "representativeEmailAddress": "Email address",
         "appealDocuments": "List of forms uploaded (Appeal form)",
         "supportingDocuments": "List of supporting documents uploaded",
-        "otherInformation": "List of other documents uploaded",
+        "otherInformation": "Additional document",
```
Comment (gmmagruder) [inline src/main/resources/check-your-answers/translation.json:30]: Welsh translation?

File: src/main/steps/edge-case/upload-other-information/content.ts [hunk 3]
```diff
@@ -5,15 +5,10 @@ import { ResourceReader } from '../../../modules/resourcereader/ResourceReader';
...
-      documentRelevance: {
-        type: 'text',
-        label: l => l.title3,
-        classes: 'govuk-text',
-        hint: h => h.line4,
-      },
       additionalInformation: {
         type: 'textarea',
         classes: 'govuk-textarea',
+        autocomplete: 'on',
```
Comment (gmmagruder) [inline src/main/steps/edge-case/upload-other-information/content.ts:11]: Why was this change made?

File: src/test/unit/utils/mockUserCase.ts [hunk 4]
```diff
@@ -58,6 +58,40 @@ export const mockUserCase3 = {
   applicantLastNames: 'Bob',
 } as unknown as any;
 
+export const mockUserCase4 = {
```
Comment (gmmagruder) [inline src/test/unit/utils/mockUserCase.ts:61]: Where are these new constants used?

File: src/main/steps/edge-case/upload-other-information/content.ts (hunk 3 above)
Comment (pr00279) [reply src/main/steps/edge-case/upload-other-information/content.ts]: A failed attempt to make it so that the input into the additionalInformation would not be erased on file upload; another solution has been coded into the template.njk for the additional documents page now

File: src/main/resources/check-your-answers/translation.json (hunk 2 above)
Comment (pr00279) [reply src/main/resources/check-your-answers/translation.json:30]: Thank you, has been added now

**Comment by tomxelliott** [comment]
Spacing in the CYA page for the docs looks off - they are not aligned.

**Comment by tomxelliott** [comment]
Document relevance and test1/test2 are not aligned

**Comment by pr00279** [comment]
> Document relevance and test1/test2 are not aligned

Resolved - see description with updated screenshot

File: src/main/app/form/Form.ts [hunk 5]
```diff
@@ -119,7 +119,7 @@ export type FormFields = Record<string, FormField>;
 export type FormFieldsFn = (userCase: Partial<Case>, language?: string) => FormFields;
 
 export interface FormContent {
-  submit: {
+  submit?: {
```
Comment (tomxelliott) [inline src/main/app/form/Form.ts:122]: why is ? added to the interface?

Comment (pr00279) [reply src/main/app/form/Form.ts:122]: This is to allow the additional information to be included in one form whilst allowing the other form to be submitted without a submit button (as this is a hidden submitted value - I did not remove it altogether as this would be out of scope and may be relevant in future)

Comment (pr00279) [reply src/main/app/form/Form.ts:122]: (This prevents the additional information from being removed when the 'Upload' button is clicked and ensures the value is saved and passed to the CYA page)

**Review by tomxelliott — DISMISSED** [review]

**Review by oleon23 — APPROVED** [review]

**Review by gmmagruder — APPROVED** [review]

**Review by tomxelliott — APPROVED** [review]

File: src/main/steps/edge-case/check-your-answers/utils.ts [hunk 6]
```diff
@@ -251,22 +251,22 @@ export const OtherInformationSummary = (
     changeUrl: Urls['UPLOAD_OTHER_INFORMATION'],
   };
 
-  const documentInformation = OtherDocuments.map((document: DocumentUpload) => {
-    return {
+  const docInfoArray: { [key: string]: string }[] = [];
```
Comment (tomxelliott) [inline src/main/steps/edge-case/check-your-answers/utils.ts:254]: happy for this to still be called documentInformation

**Review by tomxelliott — APPROVED** [review]

**Review by gmmagruder — APPROVED** [review]

**Review by tomxelliott — DISMISSED** [review]

**Review by oleon23 — DISMISSED** [review]

**Review by tomxelliott — APPROVED** [review]
